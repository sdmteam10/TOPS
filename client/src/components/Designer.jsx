import React, { Component } from 'react';

/*This is the designer's landing page. */
class Designer extends Component {
  render() {
    return (

      
      <div>
        <div className="mt-md-5 ws-0 mx-md-auto">
            <div>
            <h1 className="text-secondary fw-light font-monospace">Welcome Designer!</h1>
            <br />
            <br />
            </div>
            <div>
            <a className="link-secondary fw-light font-monospace" href="OrgInfo">Organisation Information Form</a>
            </div>
            <div>
            <a className="link-secondary fw-light font-monospace" href="TeamInfo">Team Information Form</a>
            </div>
            <div>
            <a className="link-secondary fw-light font-monospace" href="Goals">Goals Page</a>
            </div>
        </div>
      </div>
    )
  }
}

export default Designer