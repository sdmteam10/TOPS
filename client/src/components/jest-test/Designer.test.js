// Designer.test.js
import React from 'react'
import Designer from '../Designer'
import renderer from 'react-test-renderer';
import { fireEvent, render } from "@testing-library/react"

//it.only() "only test this <it>."
//it.skip() "not test this <it>."
// done() under expect(){} -  should be only used for async functions

  describe('Test <Designer /> component', function() {
    it('Should render', function() {  
      //Render component should be successed.
      const component = renderer.create(
        <Designer />
      );
        let testSnap = component.toJSON();
        expect(testSnap).toMatchSnapshot(); 
    })
  })