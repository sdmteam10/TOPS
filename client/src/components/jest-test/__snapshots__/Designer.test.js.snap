// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Designer /> component Should render 1`] = `
<div>
  <div
    className="mt-md-5 ws-0 mx-md-auto"
  >
    <div>
      <h1
        className="text-secondary fw-light font-monospace"
      >
        Welcome Designer!
      </h1>
      <br />
      <br />
    </div>
    <div>
      <a
        className="link-secondary fw-light font-monospace"
        href="OrgInfo"
      >
        Organisation Information Form
      </a>
    </div>
    <div>
      <a
        className="link-secondary fw-light font-monospace"
        href="TeamInfo"
      >
        Team Information Form
      </a>
    </div>
    <div>
      <a
        className="link-secondary fw-light font-monospace"
        href="GoalsActivitiesList"
      >
        Activity Scheduler
      </a>
    </div>
  </div>
</div>
`;
